### DISCUSSION TITLE
Lob und Frage

### DISCUSSION THREAD

**Discussion by HenningS81** [xgcssch/SystaComfortPrometheusExporter#5] [General]
Hallo Sönke,

vielen Dank für Deine tolle Arbeit und die super Anleitung, nach der ich es sofort geschafft habe, alles auf meinem Synology NAS einzurichten. Auch dem DNS Server konnte ich für diese eine Adresse die neue IP Adresse verklickern.
Nun fehlt mir nur noch die Geschwätzigkeit meiner Anlage zum meinem Glück.  Da ich die Anlage mit dem Haus gekauft habe, fehlt mir der Kontakt zu dem Installateur. Hast Du da einen, der die Software rausgibt?

Danke und viele Grüße,

Henning

**Comment by xgcssch** [comment]
Hallo Henning!
Es ist schon schade, dass Paradigma das nicht in irgendeinen Downloadbereich hat.
Kontaktiere mich doch mal unter aaryav.madix@usweek.net, dann kann ich Dir ev. jemanden nennen.
Sönke

**Comment by xgcssch** [comment]
Sorry, abgelaufene Adresse. Die hier ist eine Woche gültig: johnathan.johnanthony@usweek.net

**Comment by kaiherzig** [comment]
Ist deine E-Mail Adresse noch gültig?
Ich hab das gleiche Problem :/
Meine Anlage sendet nur über Port 8001/UDP

**Reply by xgcssch** [reply]
Hallo!
Die Mailadresse war schon ungültig.
Ich habe in der deutschen Anleitung eine kleine FAQ geschrieben.
Dort habe ich auch den Punkt aufgeführt, das man das Programm leider aus lizenzrechtlichen Gründen nur über Paradigma bzw. seinem Heizungsbauer erhalten kann.
Sorry, 
Gruß
Sönke

**Comment by tobiasfiechter** [comment]
Hi Sönke

Vielen Dank für deine Bemühungen! Ich hab vor ca. 2 Wochen versucht über Pradigma und den Heizungsmonteur an die Software zu kommen, leider ohne Erfolg, von beiden habe ich keine Antwort erhalten. Hast du direkt zu eine(r|m) Paradigma Mitarbeiter(In) kontakt?

Grüsse
Tobias

**Reply by tobiasfiechter** [reply]
Ok ich hab inzwischen von Paradigma eine Antwort erhalten. Dabei hab ich Hex Files für ein Update, ein Program für die Deaktivierun der Verschlüsselung und eine Dokumentation für das Auslesen (und Schreiben?) der Anlage über modbus tcp erhalten.
Leider fehlt mir momentan die Zeit für genauere Analysen.
 Mit der Deaktivierung der Verschlüsselung krieg ich Daten im Prometheus! :)

Vielen Dank

**Reply by kaiherzig** [reply]
Ich hab dir grade ne Mail geschrieben ;)

**Reply by HenningS81** [reply]
Hallo Tobias,

das ist doch mal eine positive Nachricht, dass Paradigma da reagiert hat.

Mit welcher Version der Firmware läuft es denn bei Dir jetzt?

Viele Grüße,

Henning

**Reply by xgcssch** [reply]
Hallo!

Die Modbus Dokumentation hatte mir Paradigma auch zugeschickt, als ich mich nach Dokumentation erkundigt habe.
Leider reagiert meine Systacomfort nicht auf Anfragen in dem darin dokumentierten Port 502:

> Die Modbus-Server akzeptieren auf Port 502 Anfragen von Modbus-Clients:

Bei einem Scan auf die unteren Ports antworten bei TCP nur 21(FTP) und 80(HTTP):
```
$ nc -zvw 1 systacomfort.int.schau.org 10-1000 2>&1 |grep succeeded
Connection to systacomfort.int.schau.org 21 port [tcp/ftp] succeeded!
Connection to systacomfort.int.schau.org 80 port [tcp/http] succeeded!
```
Bei letzterem gibt es nur einen `404` auf der Root:
```
$ curl http://systacomfort.int.schau.org
404 Not Found
```

Eigentlich sehr schade, dass das über den Modbus-TCP nicht geht. Für das Protokoll gibt es freie Bibliotheken und es würde auch die Möglichkeit bieten, die Heizung nicht nur zu monitoren, sondern auch zu steuern.
Wenn das gehen würde, könnte ich die Ansteuerung sofort umbauen und auch das Setup würde sich deutlich vereinfachen.

Ev. könnten wir ja folgendes machen:
1. Prüft doch bitte, ob der Modbus TCP Port bei Euren Anlagen offen ist und antwortet.
2. Wenn dieser nicht offen ist, befragen wir Paradigma, ob dieser für unsere Anlagen ohne zusätzliche Hard- oder Software aktivierbar ist.

@tobiasfiechter : Hättest du ev. Muße, bei Paradigma eine Rückfrage zu machen?

Lieben Gruß
Sönke

**Reply by HenningS81** [reply]
Hallo Sönke,

bei mir sind auch nur die beiden Ports erreichbar.

LG,
Henning

**Reply by HenningS81** [reply]
Die Programme sind hier tatsächlich frei verfügbar: https://www.paradigma.de/software/

**Reply by xgcssch** [reply]
Hallo Henning!
Super, dass du die Stelle gefunden hast, wo die Software heruntergeladen werden kann!
Ich habe das auch gleich in die FAQ aufgenommen.
Außerdem habe ich eben noch eine Mail an Paradigma geschrieben und gefragt, ob die Modbus-TCP Anbindung in unseren Anlagen überhaupt vorhanden sein sollte, oder ob diese irgendwie freigeschaltet werden muss.
Mal sehen, was die antworten.
LG Sönke

**Reply by HenningS81** [reply]
Hi Sönke,

ich habe jetzt mal auf die 1.26 geantwortet, bei den Ports hat sich aber nichts getan.

LG Henning

**Reply by xgcssch** [reply]
Schade!
Ich habe leider auch noch keine Antwort von Paradigma.
Nächste Woche hake ich da nochmal nach.
Sönke

**Comment by tobiasfiechter** [comment]
Ich hab nicht viel Zeit im Moment, aber folgende Infos hab ich von Paradigma erhalten:

Testsoftware SystaComfort II, Applikation 2.15.8, Basissystem 0.51.7

Schritt 1: Falls schon eine SD-Karte im Regler ist, mir bitte die Daten mailen. Daten können auf der SD bleiben.
Schritt 2: Ordner Firmware auf einer SD-Karte anlegen, die beiden angehängten .HEX Dateien in den Ordner legen. Der Ordner Firmware soll keine weiteren Dateien enthalten. Die schon im Regler vorhandene SD kann verwendet werden. Wichtig: Die neue Applikation funktioniert nur mit dem neuen Basissystem (und umgekehrt) – es müssen immer beide Updates gemacht werden.
Schritt 3: SD-Karte in den Regler stecken.
-  Anzeige Update auf V0.51.7 ? bestätigen
-  Regler macht 2 Updates nacheinander (Basissystem und Applikation). Dann werden noch neue Daten in die Bedienteile S-Touch geladen. Wenn alle Bedienteile wieder den Standardbildschirm zeigen, ist das Update abgeschlossen.
Schritt 4: Prüfen: Im Menü System > Kontrolle müssen als Softwareversionen stehen: - Basis-System: 0.51.7
- Applikation: 2.15.8
Schritt 5: 2-3 Minuten warten, bis der Regler alle Erweiterungen und Kessel richtig erkannt hat. Währenddessen können kurzzeitig Störmeldungen erscheinen (einige Sekunden).
Schritt 6: Wenn möglich kurz prüfen, ob der Kessel korrekt eingeschaltet wird, zum Beispiel Warmwassertemperatur hochstellen und Jetzt aufheizen drücken
Schritt 7: Systemlog einschalten:
-  Menü System > Anlagendaten > Systemlog auf SD-Karte aufrufen (Menü Anlagendaten ist nur sichtbar bei aktivem Code 21 !)
-  180 Tage einstellen. Nach 180 Tagen wird der Log automatisch ausgeschaltet.
- Der Systemlog ist neu in dieser Version und enthält für uns sehr hilfreiche Informationen bei Netzwerkproblemen und Regler-Abstürzen.
Schritt 8: System > Anlagendaten > täglicher Netzwerk-Reset ausschalten, falls aktiv
Schritt 9: System > Kontrolle > Status SystaWeb prüfen Falls keine Verbindung: Einsteller System > Einstellungen > Datenübertragung SystaWeb prüfen, muss auf Erlaubt stehen
Schritt 10: SD-Karte im Regler lassen

**Reply by tobiasfiechter** [reply]
Es sieht so aus, als muss man den modbus Server aktivieren.

**Reply by xgcssch** [reply]
Stimmt, das ist die entscheidende Frage.
Im Prinzip läuft es auf die Frage hinaus:
kann man bei unseren Anlagen die Modbus Option aktivieren/nachinstallieren oder ist das eine nicht enthaltene Option, welche nur mit einer kostenpflichtigen Hardware-/Softwareaufrüstung zur Verfügung stelt.
Ich habe auch noch nichts von Paradigma zu meiner Anfrage dazu gehört.
Ev. werde ich nächste Woche mal nachfragen.
LG Sönke

**Reply by jos-webservices** [reply]
Modbus wird im Zusammenhang mit bestimmten Pellets-Kesseln genutzt. Unser Peleo Optima ist mit der SystaComfort II per ModBus verbunden. Leider kann man es sonst zu nichts nutzen, zumindest schaffe ich es nicht. Ich hab schon echt viel rumprobiert und auch Paradigma dazu angefragt. Die Antwort war: "Mit Mitteln aus unserem Haus ist ein Auslesen des ModBus nicht möglich."
Das klingt nicht ganz nach "unmöglich", aber ich hab es erstmal aufgegeben.

**Reply by S-Bil** [reply]
Bei der Paradigma Heizungsanlage ist der Master - der Heizungsregler. Er sagt dem Kessel wann er einspringen soll und mit welchen Vorlauftemperatur. Welche Informationen brauchst Du von dem Kessel? Brennraumtemperatur, Abgastemperatur, Unterdruck im Brennkamer...?
Alle Störungen werden an Comfort II weitergegeben, über Modbus kann man alles ablesen und automatisiere. Z.B Alexa Ansage "Störung Kessel, bitte verlassen Sie das Haus")))

**Reply by jos-webservices** [reply]
@S-Bil das klingt interessant. Welche Werte genau ich gerne hätte, muss ich nochmal zusammenschreiben. Aber das wäre ja schon klasse, wenn das klappen würde. Ich bin nur bisher nicht wirklich weiter gekommen. Hast du mir mehr Infos, wo/wie ich das hinbekomme?
Ich glaub ich hatte sicherlich auch den Denkfehler, dass ich davon ausging, dass der Kessel der Master ist. 💡

**Reply by S-Bil** [reply]
Morgen,
Welchen Softwarestand hat der Comfort II Regler. Modbus Schnittstelle ist ab Version 2.16 
Anbei ist Bedienungsanleitung
[TH-3000_V1.0_0221_SystaComfort_II_CompactC_Modbus_Informationen_B_Fhw.pdf](https://github.com/xgcssch/SystaComfortPrometheusExporter/files/7531603/TH-3000_V1.0_0221_SystaComfort_II_CompactC_Modbus_Informationen_B_Fhw.pdf)

**Reply by S-Bil** [reply]
hier ist die 2.16 Version
Speicherkarte in FAT32 formatieren. 
Ordner "Firmware" auf einer SD-Karte anlegen, die beiden .HEX Dateien in den Ordner legen.
Der Ordner Firmware soll keine weiteren Dateien enthalten
[SystaComfort_II_r2_V2.16.1.zip](https://github.com/xgcssch/SystaComfortPrometheusExporter/files/7531636/SystaComfort_II_r2_V2.16.1.zip)

**Reply by jos-webservices** [reply]
Mega Service, vielen Dank! Und darüber kommt man dann auch an weitere Werte des Kessels? Ich mach heute Abend das Update und schau mal, wie weit ich komme... 👍

**Comment by jos-webservices** [comment]
Danke dir. Und schade, ich bin noch auf der 2.14. Ich schau mal, dass ich an die 2.16 rankomme.
